### DISCUSSION TITLE
Call API Get iTwin iModels

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/itwinjs-core#7017] [Q&A] [answered]
hello everyone

it's been a few days that my web app can no longer find the models related to my project

While the call for projects works correctly
https://api.bentley.com/itwins/?subClass[&status][&type][&number][&displayName][&parentId][&iTwinAccountId][&$search][&$top][&$skip][&includeInactive]

The call for models related to projects returns 404 - not found
https://api.bentley.com/imodels/?iTwinId[&$skip][&$top][&$orderBy][&$search][&name][&state]

I use a c# back-end application to connect to the data for projects and models,

I have obviously done some tests on
https://developer.bentley.com/apis/imodels-v2/operations/get-itwin-imodels/
and it works correctly

I noticed in my application that the roles have been changed and I would not want this whole problem to be related to this
<img width="591" alt="rolenew" src="https://github.com/user-attachments/assets/f893a8dc-cacf-4753-8d07-b542c120913d">

Thanks for the precious help

**Comment by austeja-bentley** [comment]
hello,
issues with calling `https://api.bentley.com/imodels` might be related to the fact that we disabled the deprecated v1 version recently. What `accept` header value are you sending with your requests? The value should be set to `application/vnd.bentley.itwin-platform.v2+json`.

**Reply by Paolo-lab** [reply]
hi tanks from response this is code c#  the v2 is ok 


      

        client.DefaultRequestHeaders.Add("Accept", "application/vnd.bentley.itwin-platform.v2+json");
        client.DefaultRequestHeaders.Add("Authorization", request.Token);

        using var response = await client.GetAsync($"{API_BASE_URL}{Firstpath}/?iTwinId={request.ProjectId}");

  
        responseMsg.Status = response.StatusCode;
        responseMsg.Content = await response.Content.ReadAsStringAsync();

**Reply by austeja-bentley** [reply]
yes, this looks correct. Could you provide the iTwin id for which you received 404 response? `X-Correlation-id` header value from such response would also be useful. Do you receive 404 consistently for this request, or are there particular iTwin ids for which you get the 404 response?

**Reply by Paolo-lab** [reply]
Hi , yes this is iTwin id 'a2a4966d-28e6-4c15-b5ee-8247a7933055' 

really thank you

**Reply by Paolo-lab** [reply]
Hi I wanted to add more information you requested, this problem occurs on all projects and not just the one I passed you

this is the header response
Headers = {x-correlation-id: aced724b-4865-4588-8a53-4698855df0d2
Request-Context: appId=cid-v1:bde9711b-b84c-4c4b-a161-2546ac448498
X-ITwinPlatform-Media-Type: bentley.itwin-platform.v1
X-ITwinPlatform-Region: UK South
WWW-Authenticate: AzureApiManagementKey rea...

as you can see (I don't know if this is important) X-ITwinPlatform-Media-Type in the response and is linked to the v1 version
X-ITwinPlatform-Media-Type: bentley.itwin-platform.v1 , while in my c# code I explicitly indicate

client.DefaultRequestHeaders.Add("Accept", "application/vnd.bentley.itwin-platform.v2+json");

thanks for the help

**Reply by austeja-bentley** [reply, marked as answer]
In our logs I can see that this request came in with the following header value:
```
Accept:
 application/vnd.bentley.itwin-platform.v1+json,application/vnd.bentley.itwin-platform.v2+json
```

It could be that before you call `client.DefaultRequestHeaders.Add("Accept", "application/vnd.bentley.itwin-platform.v2+json"`, `client.DefaultRequestHeaders` already has `Accept` header with value `application/vnd.bentley.itwin-platform.v1+json`.

Link to .NET docs:
https://learn.microsoft.com/en-us/dotnet/api/system.net.http.headers.httpheaders.add?view=net-8.0
![image](https://github.com/user-attachments/assets/81644ebc-80b2-4115-8163-b7e5be95b3a2)
